spi/src/main/java/org/opensearch/jobscheduler/spi/schedule/IntervalSchedule.java

```diff
@@ -107,22 +116,31 @@ public ChronoUnit getUnit() {
         return this.unit;
     }
 
+    public Long getDelay() { return this.scheduleDelay; }
+
     @Override
     public Instant getNextExecutionTime(Instant time) {
         Instant baseTime = time == null ? this.clock.instant() : time;
-        long delta = (baseTime.toEpochMilli() - this.startTime.toEpochMilli()) % this.intervalInMillis;
-        long remaining = this.intervalInMillis - delta;
-
-        return baseTime.plus(remaining, ChronoUnit.MILLIS);
+        long delta = (baseTime.toEpochMilli() - this.startTimeWithDelay.toEpochMilli());
+        if (delta >= 0) {
+            long remaining = this.intervalInMillis - (delta % this.intervalInMillis);
+            return baseTime.plus(remaining, ChronoUnit.MILLIS);
+        } else {
+            return this.startTimeWithDelay;
+        }
     }
 
     @Override
     public Duration nextTimeToExecute() {
-        long enabledTimeEpochMillis = this.startTime.toEpochMilli();
+        long enabledTimeEpochMillis = this.startTimeWithDelay.toEpochMilli();
         Instant currentTime = this.clock.instant();
         long delta = currentTime.toEpochMilli() - enabledTimeEpochMillis;
-        long remainingScheduleTime = intervalInMillis - (delta % intervalInMillis);
-        return Duration.of(remainingScheduleTime, ChronoUnit.MILLIS);
+        if (delta >= 0) {
+            long remainingScheduleTime = intervalInMillis - (delta % intervalInMillis);
+            return Duration.of(remainingScheduleTime, ChronoUnit.MILLIS);
+        } else {
+            return Duration.ofMillis(enabledTimeEpochMillis - currentTime.toEpochMilli());
+        }
```

Comment:
Same comment on Ternary operators! Simplification of code increases its readability.

Reply:
I am also a big fan of ternary operators! But I try to only use them in simple cases, and I really try to not include anything that takes much thought. I agree with you on the other cases but I will leave this one as an if/else for that reason